### PR TITLE
[master] Bug 518774: HANAPlatform is not detected; improving platform detection

### PR DESCRIPTION
**Pull request by dazey3**
for #518

Signed-off-by: Will Dazey <dazeydev.3@gmail.com>

### CODE REVIEW

File: jpa/eclipselink.jpa.test.jse/src/org/eclipse/persistence/jpa/test/ddl/TestDBPlatformHelper.java [hunk 1]
```diff
@@ -51,19 +51,47 @@ public void test() {
...
+    }
+
+    @Test
+    public void testOracle() {
+        //Returned from jcc driver (DRDA)
+        Assert.assertEquals("org.eclipse.persistence.platform.database.oracle.Oracle9Platform", getPlatformClass("Oracle", "9", "Oracle Database 9c"));
+        Assert.assertEquals("org.eclipse.persistence.platform.database.oracle.Oracle10Platform", getPlatformClass("Oracle", "10", "Oracle Database 10c"));
+        Assert.assertEquals("org.eclipse.persistence.platform.database.oracle.Oracle11Platform", getPlatformClass("Oracle", "11", "Oracle Database 11c"));
+        Assert.assertEquals("org.eclipse.persistence.platform.database.oracle.Oracle12Platform", getPlatformClass("Oracle", "12", "Oracle Database 12c"));
+        Assert.assertEquals("org.eclipse.persistence.platform.database.oracle.Oracle18Platform", getPlatformClass("Oracle", "18", "Oracle Database 18c"));
```
Comment (lukasj) [inline jpa/eclipselink.jpa.test.jse/src/org/eclipse/persistence/jpa/test/ddl/TestDBPlatformHelper.java]: 'org.eclipse.persistence.platform.database.oracle.OraclePlatform' is valid too, maybe wanna add it?

Comment (dazey3) [reply jpa/eclipselink.jpa.test.jse/src/org/eclipse/persistence/jpa/test/ddl/TestDBPlatformHelper.java]: Sure! What are the conditions to have OraclePlatform be set as the database platform? The only test case that comes to mind for me is when the version  < 9?

Comment (lukasj) [reply jpa/eclipselink.jpa.test.jse/src/org/eclipse/persistence/jpa/test/ddl/TestDBPlatformHelper.java]: the case used to be the fallback when either the exact version is not recognized or if 'oracle' jar is missing

Comment (dazey3) [reply jpa/eclipselink.jpa.test.jse/src/org/eclipse/persistence/jpa/test/ddl/TestDBPlatformHelper.java]: I'm not sure how to test for the latter, so I just added a test for the former. Does this work for you?

Comment (lukasj) [reply jpa/eclipselink.jpa.test.jse/src/org/eclipse/persistence/jpa/test/ddl/TestDBPlatformHelper.java]: Yes, sth like this is good enough.

BTW: NOT doing force-pushes makes it easier for others to find the exact diff between code which was already reviewed and new revision. Imagine the diff has 1k lines, one line needs to be changed and one has to verify that only that one particular line was changed...; one can use 'squash and merge' in github UI to make one commit out of the whole PR...

Comment (dazey3) [reply jpa/eclipselink.jpa.test.jse/src/org/eclipse/persistence/jpa/test/ddl/TestDBPlatformHelper.java]: I suppose that would make review easier, thank you for the suggestion.

**Pull request merged**